### ISSUE TITLE
[Site Editor] The save panel does not receive keyboard focus 

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#29803]
## Description

Using NVDA (a screen reader), the secondary Save button in the "Update Design" does not have an aria label making it very difficult to save changes in the FSE experience. A user relying on screen readers would need to both know to search for save and actually search for it in order to properly save changes.

## Step-by-step reproduction instructions

1. Setup a site with FSE using a block theme. Use NVDA. 
2. Make a few changes to the FSE experience. 
3. Select "Update Design" and notice that the "Save" button isn't announced. 

## Expected behaviour

I expect an aria label for the additional save button. 

## Actual behaviour

There is no aria label for the save button: 

<img width="927" alt="Screen Shot 2021-03-11 at 8 20 01 PM" src="https://user-images.githubusercontent.com/26996883/110887252-30836980-82a7-11eb-84a0-47e331c9801a.png">

## Screenshots or screen recording (optional)

Here's a screen recording shared with consent from the person who I worked with to go over this experience: 

https://user-images.githubusercontent.com/26996883/110887536-bbfcfa80-82a7-11eb-9c6c-97ea3b4d5f92.mov

## WordPress information
- WordPress version: 5.7
- Gutenberg version: 10.1
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? TT1 Blocks

**Comment by carolinan** [comment]
Hi
When I tested with NVDA and Chrome on Windows 10, I am found that the save panel does not receive focus when it is opened.

Steps:
I pressed enter when the Update design button was selected.
I could see that the save panel was opened. But there was no announcement, NVDA did not read out that the panel had opened.
While I could see that the panel had a save button at the top, the button did not have a visible focus style.

I pressed the arrow down key, but instead of reading out the save button, the screen reader read:
"toggle button not pressed, settings"'
-This is the button that is next to the "Update design", now covered by the open save panel.
I tested using tab instead of arrows, with the same result.

**Comment by carolinan** [comment]
There are some other inconsistencies, the post editor has a cancel button with visible text while the site editor panel has an "X" with the aria label "close panel" instead of "Cancel".

**Comment by annezazu** [comment]
Thank you for testing and for the very thorough comment <3

**Comment by carolinan** [comment]
I tried looking into it but it is not clear to me why there was a decision to use a different markup and CSS classes in the site editor compared to the post editor.
